cl/scrapers/management/commands/merge_opinion_versions.py

```diff
@@ -0,0 +1,688 @@
+            main_opinion.cluster.docket.save()
+
+        update_referencing_objects(main_opinion.cluster, version_cluster)
+        if not is_same_docket:
+            update_referencing_objects(main_docket, version_docket)
+
+        # update the cluster_id to prevent the version cluster deletion cascade
+        for version_citation in version_cluster.citations.all():
+            if str(version_citation) in main_citations:
+                continue
+            version_citation.cluster_id = main_opinion.cluster.id
+            version_citation.save()
+
+        version_opinion.cluster_id = main_opinion.cluster.id
+        version_opinion.main_version = main_opinion
+        version_opinion.save()
+
+        # Both Opinion and Citation have a ForeignKey to OpinionCluster, with
+        # on_delete=models.CASCADE. Also, there are signals (see
+        # o_cluster_field_mapping) to delete the related Elasticsearch docs
+        # So, deleting the cluster will delete any associated Citation,
+        # and will delete the associated objects in ES
+        version_cluster.delete()
+
+        if not is_same_docket:
+            version_docket.delete()
+
+        # since the cluster was reassigned, this opinion was not deleted from
+        # the ES index. We don't want versions to show up on search
+        try:
+            OpinionDocument.get(
+                id=ES_CHILD_ID(version_opinion.id).OPINION
+            ).delete()
+        except NotFoundError:
+            pass
+
+
+def merge_versions_by_download_url(
+    url_start: str, limit: int | None = None
+) -> None:
+    """Get opinion version candidates when they have the exact same download_url
+
+    Note that this function applies only to scraped Opinions, which have a
+    `download_url` field
+
+    :param url_start: the general domain and directory parts of the URL as
+        found on Opinion.download_url. For example, for `nev` opinions:
+        'http://caseinfo.nvsupreme'
+    :param limit: max number of groups to correct
+    :return None
+    """
+    if url_start:
+        query = get_query_from_url(url_start, "startswith")
+    else:
+        query = Q()
+
+    qs = (
+        Opinion.objects.filter(query)
+        .exclude(Q(download_url="") | Q(download_url__isnull=True))
+        .values("download_url")
+        .annotate(
+            number_of_rows=Count("download_url"),
+            # compute the number of  distinct hashes to prevent colliding with
+            # actual duplicates, which are not versions
+            number_of_hashes=Count("sha1", distinct=True),
+        )
+        .order_by()
+        .filter(number_of_rows__gte=2, number_of_hashes__gte=2)
+    )
+
+    # The groups queryset will look like
+    # {'download_url': 'https://caseinfo.nvsupremecourt...',
+    # 'number_of_rows': 2, 'number_of_hashes': 2}
+    seen_urls = set()
+    stats = defaultdict(lambda: 0)
+    for group in qs:
+        if group["download_url"].replace("https", "http") in seen_urls:
```

Comment:
maybe put `group["download_url"].replace("https", "http")` into a variable to reuse it again in L538